app/models/compute.rb

```diff
@@ -0,0 +1,82 @@
+class Compute
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+  include ActiveModel::Dirty
+
+  attr_accessor :compute_resource, :host
+
+  attribute :uuid, :string
+  attribute :name, :string, default: -> { "foreman_#{Time.now.to_i}" }
+
+  def initialize(*attrs)
+    super
+    uuid ||= host.uuid
+    @attrs_loaded = false
+    raise 'host and compute_resource need to be set for Compute' unless host && compute_resource
```

Comment:
Id move this so it happens before you use `host.uuid` to avoid a `nil.uuid` call.